tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -171,6 +174,640 @@ TILEDB_EXPORT int32_t tiledb_array_evolve(
 TILEDB_EXPORT int32_t tiledb_array_upgrade_version(
     tiledb_ctx_t* ctx, const char* array_uri, tiledb_config_t* config);
 
+/* ********************************* */
+/*              FILE                 */
+/* ********************************* */
+
+/**
+ * Allocs the tiledb_file_t type
+ *
+ * @param ctx The TileDB context.
+ * @param array_uri the uri of the array.
+ * @param file The tiledb_file_t to be allocated.
+ * @param config Configuration parameters for the upgrade.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_alloc(
+    tiledb_ctx_t* ctx, const char* array_uri, tiledb_file_t** file);
+
+/**
+ * Sets the file config.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_open(ctx, file, TILEDB_READ);
+ * // Set the config for the given file.
+ * tiledb_config_t* config;
+ * tiledb_file_set_config(ctx, file, config);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file to set the config for.
+ * @param config The config to be set.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ *
+ * @note The file does not need to be opened via `tiledb_file_open` to use
+ *      this function.
+ * @note The config should be set before opening an file.
+ */
+TILEDB_EXPORT int32_t tiledb_file_set_config(
+    tiledb_ctx_t* ctx, tiledb_file_t* file, tiledb_config_t* config);
+
+/**
+ * Gets the file config.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * // Retrieve the file for the given array.
+ * tiledb_config_t* config;
+ * tiledb_file_get_config(ctx, file, config);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The file to set the config for.
+ * @param config Set to the retrieved config.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_get_config(
+    tiledb_ctx_t* ctx, tiledb_file_t* file, tiledb_config_t* config);
+
+/**
+ *
+ * Destroys an file object, freeing associated memory.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_free(&file);
+ * @endcode
+ * @param file The file object to destroy
+ */
+TILEDB_EXPORT void tiledb_file_free(tiledb_file_t** file);
+
+/**
+ * Create a file array with default schema
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_file_t* file;
+ * tiledb_file_alloc(ctx, "s3://tiledb_bucket/my_file", &file);
+ * tiledb_file_create_default(ctx, file);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param file The tiledb_file_t to be created.
+ * @param config Configuration parameters for the upgrade.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_file_create_default(
```

Comment:
I think that this should just take as input the file URI, not a `tiledb_file_t` object, similar to what we do with `tiledb_array_create`.